### ISSUE TITLE
svmap: System.NullReferenceException

### ISSUE THREAD

**Issue by l3owie** [matigramirez/Parsec#31]
**Describe the bug**
System.NullReferenceException thrown when trying to build an svmap file from JSON.

**To Reproduce**
1. Set a debug breakpoint at line 50 in Reader.cs and step into the code
2. Keep stepping until it passes through the empty FileBase constructor
3. Once it reaches the Monster constructor it will throw the exception

```
System.NullReferenceException: 'Object reference not set to an instance of an object.'

binaryReader was null.
```
```
System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=Parsec
  StackTrace:
   at Parsec.Shaiya.Svmap.Monster..ctor(SBinaryReader binaryReader) in C:\Users\kurtekat\source\repos\Parsec\src\Parsec\Shaiya\Svmap\Monster.cs:line 15
```

**Desktop**
 - Windows 7 Home Premium
 - Visual Studio 2019 Community Edition

**Additional context**
```c#
using Parsec.Readers;
using Parsec.Shaiya.Svmap;

namespace ConsoleApp
{
    class Program
    {
        static void Main(string[] args)
        {
            var svmap = Reader.ReadFromJson<Svmap>("0.json");
            svmap.Write("0.svmap");
        }
    }
}
```

**Comment by matigramirez** [comment]
Good catch. This issue was caused by all of the `svmap` subclasses not having parameterless constructors for json to use. They instead were being created with the ctor that attempts to use `SBinaryReader` which was null.
Issue was fixed on [eeb838072171efedb8f9deb3fab4955eaa37d3d3](https://github.com/matigramirez/Parsec/commit/eeb838072171efedb8f9deb3fab4955eaa37d3d3).

**Comment by l3owie** [comment]
glad i was able to help. thanks for fixing it.
